guides/workflow/testing.md

```diff
@@ -0,0 +1,197 @@
+# Testing
+
+Testing is essential for creating and maintaining your application code. With Wix CLI, you can easily use popular tools like Jest and Vitest for component tests.
+
+These testing frameworks let you define expected behaviors for your code in different scenarios and then compare them with the actual outcomes.
+
+In the sections below, we'll show you how to set up specific testing frameworks with Wix CLI.
+
+## Vitest
+
+Vitest is a blazing fast unit test framework powered by Vite ([Learn more about Vitest](https://vitest.dev/guide/why.html)).
+
+Following Vitest's [getting started guide](https://vitest.dev/guide/), you can start by installing Vitest:
+
+```
+yarn add --dev vitest
+```
+
+Also, install `jsdom` which we'll need for simulating a browser-like environment:
+
+```
+yarn add --dev jsdom
+```
+
+Then, create the following `vitest.config.mjs` file at the root of your application:
+
+```javascript
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    environment: 'jsdom',
+  },
+});
+```
+
+Next, add the following script to your `package.json`:
+
+```diff
+{
+    "scripts": {
++     "test": "vitest"
+    }
+}
+```
+
+We recommend you use [Testing Library](https://testing-library.com/) for testing your UI components. To set it up, start by installing it:
+
+```
+yarn add --dev @testing-library/jest-dom @testing-library/react@^12.0.0
```

Comment:
@ohana54 I made one more change to this, notice that I pinned `@testing-library/react`'s major version to 12. This is due to [version 13 not supporting react <18](https://github.com/testing-library/react-testing-library/releases/tag/v13.0.0), and even with the `legacyRoot` option there are still warnings. The generator PR is already pinned to having version ^12 in its `package.json`.